### PR TITLE
Compiling and running on arm64 macOS

### PR DESCRIPTION
**Pull request by benbeshara**
Added necessary changes to makefile
Added unistd include
Fixed segfault in qrad3

Not sure if the `trace.c` fix is a good solution or not, I think the original line was saturating all the bits in the buffer? But that's just a guess.

### CODE REVIEW

File: src/qrad3/trace.c [hunk 1]
```diff
@@ -85,7 +85,7 @@ void MakeTnodes (dmodel_t *bm)
 {
 	// 32 byte align the structs
 	tnodes = malloc( (numnodes+1) * sizeof(tnode_t));
-	tnodes = (tnode_t *)(((int)tnodes + 31)&~31);
+  memset(tnodes, 1, sizeof tnodes);
```
Comment (Yamagi) [inline src/qrad3/trace.c]: Indention is wrong. Otherwise, I'm not 100% sure either if that code is really equivalent. What an ugly hack...

Comment (turol) [reply src/qrad3/trace.c]: It definitely isn't. The original changes the pointer to align it, the new one sets the contents of the memory.

Comment (turol) [reply src/qrad3/trace.c]: BTW the proper way to do this is either `posix_memalign` or `aligned_alloc`. Annoyingly this is not entirely standard, at least until newer C/C++ versions.

Comment (benbeshara) [reply src/qrad3/trace.c]: Ah that actually makes a lot more sense than whatever I was thinking - I've attempted another fix and fixed up code the alignment, I'll test it on my PC at some point and see if it holds up :)

File: src/qrad3/trace.c [hunk 2]
```diff
@@ -84,8 +84,7 @@ Loads the node structure out of a .bsp file to be used for light occlusion
 void MakeTnodes (dmodel_t *bm)
 {
 	// 32 byte align the structs
-	tnodes = malloc( (numnodes+1) * sizeof(tnode_t));
-	tnodes = (tnode_t *)(((int)tnodes + 31)&~31);
+	tnodes = aligned_alloc(32,(numnodes+1) * sizeof(tnode_t));
```
Comment (benbeshara) [inline src/qrad3/trace.c:87]: I was way overthinking this, but this should just be
```
tnodes = malloc( (numnodes+1) * sizeof(tnode_t));
tnodes = (tnode_t *)(((intptr_t)tnodes + 31)&~31);
```
but the radiosity engine seems to produce nothing at all - just a black, empty lightmap, and I'm not sure where to go from here so I'll close this PR

**Pull request closed without merging**